pkg/frontend/transport/handler.go

```diff
@@ -190,6 +192,20 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	activityIndex := f.at.Insert(func() string { return httpRequestActivity(r, r.Header.Get("User-Agent"), params) })
 	defer f.at.Delete(activityIndex)
 
+	if isStreamingEndpoint(r) {
+		// Set write deadline for streaming responses.
+		var deadline time.Time
+		if f.cfg.ResponseStreamTimeout > 0 {
+			deadline = time.Now().Add(f.cfg.ResponseStreamTimeout)
```

Comment:
I wouldn't set a no deadline if `ResponseStreamTimeout` is 0. On the contrary, the whole `if isStreamingEndpoint(r)` should be skipped if `ResponseStreamTimeout == 0` and then we document that if it's zero the global timeout will be used. I think it's a bad practice allowing to have no timeout (even if someone could set a very high timeout, but at least it's more clear the intention).

Reply:
I have changed the behaviour to apply the longer timeout only if it's configured and otherwise default to the same timeout as the other endpoints.